File: srf-core/src/main/java/org/reggsoft/srfcore/authentication/AuthController.java
package org.reggsoft.srfcore.authentication;

import org.reggsoft.srfcore.persistance.dao.ScadaUserRepository;
import org.reggsoft.srfcore.persistance.entity.ScadaUser;
import org.reggsoft.srfcore.services.ScadaUserService;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

@Controller
@RestController
@RequestMapping(path = "api")
public class AuthController {
    private final AuthenticationManager authenticationManager;
    private final JwtTokenUtil jwtTokenUtil;
    private final ScadaUserService userService;


    public AuthController(AuthenticationManager authenticationManager,
                          JwtTokenUtil jwtTokenUtil, ScadaUserService userService) {
        this.authenticationManager = authenticationManager;
        this.jwtTokenUtil = jwtTokenUtil;
        this.userService = userService;
    }

    @PostMapping("login")
    public ResponseEntity<String> login(@RequestParam(value = "username") String username, @RequestParam(value = "password") String password) {
        try {
            Authentication authentication = authenticationManager.authenticate(
                    new UsernamePasswordAuthenticationToken(username, password)
            );
            User user = (User) authentication.getPrincipal();
            return ResponseEntity.ok().header(HttpHeaders.AUTHORIZATION, jwtTokenUtil.generateAccessToken(user.getUsername())).body("Hi");
        } catch (BadCredentialsException ex) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
        }
    }

    @PostMapping("create")
    public ResponseEntity<UserDetails> createUser(@RequestBody ScadaUser user) {
        return new ResponseEntity<>(userService.saveUser(user), HttpStatus.CREATED);
    }


}

File: srf-core/src/main/kotlin/org/reggster/srfcore/security/AuthController.kt
package org.reggster.srfcore.security

import org.reggster.srfcore.security.jwt.JWTTokenUtils
import org.reggster.srfcore.security.acl.ScadaUserService
import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.core.userdetails.User
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@Controller
@RestController
@RequestMapping("/api/v1/")
class AuthController(
    private val authenticationManager: AuthenticationManager,
    private val scadaUserService: ScadaUserService,
    private val jwtTokenUtils: JWTTokenUtils
) {

    @PostMapping("login")
    fun login(@RequestParam(value = "username") username: String, @RequestParam(value = "password") password: String): ResponseEntity<String> {
        return try {
            val auth:Authentication = authenticationManager.authenticate(
                UsernamePasswordAuthenticationToken(username, password)
            )
            val user: User = auth.principal as User
            ResponseEntity.ok().header(HttpHeaders.AUTHORIZATION, jwtTokenUtils.generateAccessToken(username)).body("HI")
        } catch (e: BadCredentialsException) {
            ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
        }
    }
}